compiler/codegen/library.cpp

```diff
@@ -232,12 +238,39 @@ static void printMakefileLibraries(fileinfo makefile, std::string name) {
           libname.c_str());
 
   //
-  // The ZMQ library has to be linked against the C++ stdlib. Rather than
-  // package libc++ up as part of our multi-locale library client when
-  // packaging it up, we'll let library-makefile do it for us.
+  // Multi-locale libraries require some extra libraries to be linked in order
+  // to function correctly. For static libraries in particular, rather than
+  // try to link these dependencies at compile time, we shunt responsibility
+  // off to the user via use of `--library-makefile`.
   //
   if (fMultiLocaleInterop) {
-    fprintf(makefile.fptr, " %s", "-lc++");
+
+    //
+    // Get the path to the launcher library.
+    //
+    std::string launcherLib = getCompilelineOption("launcher-libdir");
+    removeTrailingNewline(launcherLib);
+
+    fprintf(makefile.fptr, " -L%s", launcherLib.c_str());
+
+    fprintf(makefile.fptr, " %s", "-lchpllaunch");
+
+    //
+    // For right now, we assume GNU/Linux and use "libstdc++" unless we are
+    // on Macs, in which case we will try to link against "libc++" instead as
+    // it is the preferred default.
```

Comment:
I'm curious how PrgEnv-cray and PrgEnv-pgi respond to this, but would rather open an issue than hold up the PR if it doesn't work

Reply:
Yeah, this is definitely something that is going to have to be more fine tuned as we go along. Ultimately, though, the key here is that we link against the C++ standard library implementation that is required by ZMQ.

Reply:
Note that I expect PrgEnv-intel to follow PrgEnv-gnu's lead (but we should verify that too)